File: src/main/java/com/drunkensoftball/api/auth/basic/repo/BasicAuthenticationRepository.java
package com.drunkensoftball.api.auth.basic.repo;

import com.drunkensoftball.api.auth.basic.domain.BasicAuthentication;
import org.springframework.data.jpa.repository.JpaRepository;

public interface BasicAuthenticationRepository  extends JpaRepository<BasicAuthentication, Long> {

    BasicAuthentication findByUserUsername(String username);
}

File: src/main/kotlin/com/drunkensoftball/api/auth/repo/BasicAuthenticationRepository.kt
package com.drunkensoftball.api.auth.repo

import com.drunkensoftball.api.auth.domain.BasicAuthentication
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface BasicAuthenticationRepository : JpaRepository<BasicAuthentication, Long> {

    fun findByUserUsername(username: String): BasicAuthentication
}
